server/py/services/api/crud/alert_activation.py

```diff
@@ -15,13 +15,71 @@
 
 from datetime import datetime, timedelta
 
+import sqlalchemy.orm
+
 import mlrun.common.schemas.alert
 import mlrun.utils.singleton
+import services.api.utils.singletons.db
 
 
 class AlertActivation(
     metaclass=mlrun.utils.singleton.Singleton,
 ):
+    def store_alert_activation(
+        self,
+        session: sqlalchemy.orm.Session,
+        alert_data: mlrun.common.schemas.AlertConfig,
+        event_data: mlrun.common.schemas.Event,
+    ):
+        notifications_states = self._prepare_notifications_states(alert_data)
+
+        services.api.utils.singletons.db.get_db().store_alert_activation(
+            session, alert_data, event_data, notifications_states
+        )
+
+    def list_alerts_activations(
+        self,
+        session: sqlalchemy.orm.Session,
+        project: str = "",
+    ) -> list[mlrun.common.schemas.AlertActivation]:
+        # add filters later
+        project = project or mlrun.mlconf.default_project
+        return services.api.utils.singletons.db.get_db().list_alerts_activations(
+            session, project
+        )
+
+    @staticmethod
+    def _prepare_notifications_states(
+        alert: mlrun.common.schemas.AlertConfig,
+    ) -> list[mlrun.common.schemas.NotificationState]:
+        # process the notifications associated with the provided alert and construct a list of NotificationState objects
+        # each NotificationState represents a unique type of notification (e.g., "slack", "git") and its status,
+        # the status will be set to "error" if at least one notification of that type has failed.
+        # otherwise, the status will be an empty string if all notifications of that type have succeeded.
+        notification_states = {}
+
+        for alert_notification in alert.notifications:
+            notification_kind = alert_notification.notification.kind
+            notification_status = alert_notification.notification.status
+
+            if notification_kind not in notification_states:
+                notification_states[notification_kind] = (
+                    "error"
+                    if notification_status
+                    == mlrun.common.schemas.NotificationStatus.ERROR
+                    else ""
+                )
+            else:
+                if notification_status == mlrun.common.schemas.NotificationStatus.ERROR:
+                    notification_states[notification_kind] = "error"
```

Comment:
The same thing but a bit more efficient:
```suggestion
            notification_states.setdefault(
                notification_kind,
                notification_status if notification_status == mlrun.common.schemas.NotificationStatus.ERROR else ""
            )
```